### PR TITLE
test: e2e and unit test setup

### PR DESCRIPTION
**Pull request by AntonOlkhovskyi**
#### Please provide a link to the associated issue.
https://github.com/SAP/fundamental-ngx/issues/3389
#### Please provide a brief summary of this pull request.
Add e2e and unit tests run on ci
#### Please check whether the PR fulfills the following requirements

- [x] the commit message follows the guideline:
https://github.com/SAP/fundamental-ngx/blob/master/CONTRIBUTING.md
- [x] tests for the changes that have been done
- [x] all items on the PR Review Checklist are addressed :
https://github.com/SAP/fundamental-ngx/wiki/PR-Review-Checklist

Documentation checklist:
- [x] update `README.md`
- [x] [Breaking Changes Wiki](https://github.com/SAP/fundamental-ngx/wiki/Breaking-Changes)
- [x] Documentation Examples
- [x] Stackblitz works for all examples

### CODE REVIEW

**Review by fkolar — APPROVED** [review]
Pretty cool having this now

File: e2e/wdio/platform/tests/checkbox.e2e-spec.ts [hunk 1]
```diff
@@ -0,0 +1,346 @@
...
+            }
+        });
+    });
+
+    describe('Check Tristate Checkbox With Value Property and Without Value Property', () => {
+        it('should check reactive form', () => {
+            if (browser.capabilities.browserName === 'Safari' || 'internet explorer') {
+                console.log('skip');
+            } else {
+                for (let i = 0; 8 > i; i++) {
```
Comment (valorkin) [inline e2e/wdio/platform/tests/checkbox.e2e-spec.ts:129]: magic numbers?

**Review by valorkin — CHANGES_REQUESTED** [review]
left some comments

File: e2e/test-accessibility.e2e-spec.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+describe('Test run accessibility', function() {
```
[the remark this answers is not in the document]
Comment (AntonOlkhovskyi) [reply e2e/test-accessibility.e2e-spec.js:1]: Removed.

File: e2e/wdio/helper/assertion-helper.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+export function checkRtlOrientation(element: string, index: number): void {
+    expect(webDriver.getAttributeByName(element, 'dir', index)).toBe('rtl');
+    expect(webDriver.getCSSPropertyByName(element, 'direction', index).value).toBe('rtl');
+}
+
+export function checkLtrOrientation(element: string, index: number): void {
+    expect(webDriver.getAttributeByName(element, 'dir', index)).toBe('ltr');
+    expect(webDriver.getCSSPropertyByName(element, 'direction', index).value).toBe('ltr');
+}
+
```
[the remark this answers is not in the document]
Comment (AntonOlkhovskyi) [reply e2e/wdio/helper/assertion-helper.ts:40]: fixed

File: e2e/wdio/driver/wdio.ts [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+        // @ts-ignore
+        $$(selector)[index].focus();
+    }
+
+    clickNextElement(selector: string, index: number = 0): void {
+        $$(selector)[index].nextElement().click();
+    }
+}
+
+export const webDriver = new Wdio();
```
[the remark this answers is not in the document]
Comment (AntonOlkhovskyi) [reply e2e/wdio/driver/wdio.ts:217]: Noted. We will get read of it once WIP PR's written it the same style are merged. Is it ok?

File: e2e/wdio/platform/tests/checkbox.e2e-spec.ts [hunk 5]
```diff
@@ -0,0 +1,346 @@
...
+            webDriver.waitForDisplayed(checkboxPage.binaryTempCheckbox, 0);
+            // check checkbox labels
+            for (let i = 0; 3 > i; i++) {
+                checkIfDisabled(checkboxPage.binaryTempCheckbox, 'ng-reflect-is-disabled', 'false', i);
+                checkMarkingCheckbox(checkboxPage.binaryTempCheckbox, i);
+            }
+            if (browser.capabilities.browserName === 'Safari' || 'internet explorer') {
+                console.log('skip');
+            } else {
+                checkHoverState(checkboxPage.binaryTempCheckbox);
```
[the remark this answers is not in the document]
Comment (AntonOlkhovskyi) [reply e2e/wdio/platform/tests/checkbox.e2e-spec.ts:29]: It this case it is the best effort to skip the only mouse hover check and keep the checks that can be performed

File: e2e/wdio/platform/tests/checkbox.e2e-spec.ts [hunk 6]
```diff
@@ -0,0 +1,346 @@
...
+            }
+            checkFocusState(checkboxPage.binaryTempCheckbox);
+        });
+
+        it('should check binary checkbox in reactive/model driven form', () => {
+            if (browser.capabilities.browserName === 'Safari' || 'internet explorer') {
```
[the remark this answers is not in the document]
Comment (AntonOlkhovskyi) [reply e2e/wdio/platform/tests/checkbox.e2e-spec.ts:35]: fixed.

File: e2e/wdio/platform/tests/checkbox.e2e-spec.ts [hunk 7]
```diff
@@ -0,0 +1,346 @@
...
+                console.log('skip');
```
[the remark this answers is not in the document]
Comment (AntonOlkhovskyi) [reply e2e/wdio/platform/tests/checkbox.e2e-spec.ts:36]: fixed

File: e2e/wdio/platform/tests/checkbox.e2e-spec.ts (hunk 1 above)
Comment (AntonOlkhovskyi) [reply e2e/wdio/platform/tests/checkbox.e2e-spec.ts:129]: Some components have lots of examples with exec same locators. So it is two ways to identify them by index or by exact locations. In this case, the index looks more stable (less readable). We will think how to optimize that in the future(by adding the ids or unique attr)

File: e2e/wdio/driver/wdio.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (valorkin) [reply e2e/wdio/driver/wdio.ts:217]: ok

File: e2e/wdio/platform/tests/checkbox.e2e-spec.ts (hunk 5 above)
[the remark this answers is not in the document]
Comment (valorkin) [reply e2e/wdio/platform/tests/checkbox.e2e-spec.ts:29]: ok

File: e2e/wdio/platform/tests/checkbox.e2e-spec.ts [hunk 8]
```diff
@@ -0,0 +1,346 @@
+import { CheckboxPO } from '../pages/checkbox.po';
+import checkboxGPData from '../../platform/fixtures/appData/checkbox-page-contents';
+import checkboxData from '../fixtures/appData/checkbox-page-contents';
+import { webDriver } from '../../driver/wdio';
+
+describe('Checkbox test suite', () => {
+    const checkboxPage = new CheckboxPO();
```
[the remark this answers is not in the document]
Comment (AntonOlkhovskyi) [reply e2e/wdio/platform/tests/checkbox.e2e-spec.ts:27]: done.

**Review by valorkin — APPROVED** [review]

File: wdio.conf.js [hunk 9]
```diff
@@ -173,7 +173,7 @@ exports.config = {
     // with `/`, the base url gets prepended, not including the path portion of your baseUrl.
     // If your `url` parameter starts without a scheme or `/` (like `some/path`), the base url
     // gets prepended directly.
-    baseUrl: 'https://sap.dev:4200/',
+    baseUrl: 'https://anton.local:4200/',
```
Comment (Seamoo13) [inline wdio.conf.js:176]: sap.dev:4200, please :)

**Review by Seamoo13 — APPROVED** [review]

**Pull request merged**